### ISSUE TITLE
DEPR: Enforce 2.0 deprecations

### ISSUE THREAD

**Issue by datapythonista** [pandas-dev/pandas#50579]
xref #30228

List of deprecations pending enforcement before releasing pandas 2.0.

- #35967
- #38544
- #43028
- #45574
- #49560 (#42717)
- #49715

Those are from unchecked items in #30228 and https://github.com/pandas-dev/pandas/pull/50196#issuecomment-1371206851. We should probably run a grep to see what else is pending before we start merging PRs that introduce new deprecations (see #50578).

**Comment by jbrockmendel** [comment]
Is this redundant with #30228?

**Comment by mroeschke** [comment]
Yeah it appears these are essentially the "last" depercations that need to be enforced in https://github.com/pandas-dev/pandas/issues/30228

**Comment by rhshadrach** [comment]
https://github.com/pandas-dev/pandas/pull/35967 can be removed since it was never merged, yea?

**Comment by datapythonista** [comment]
> Is this redundant with #30228?

Yes, I thought it'd make it easier to track things. But on a second thought, I think it's probably better to simply have the list of blockers in #46776, whether they're a deprecation or not. So, closing here in favor of the RLS 2.0 issue. Sorry for the noise.

**Comment by topper-123** [comment]
What is the status of #35967 wrt. changing the default for the `observed` in groupby? The parameter default was never deprecated, so that means it will not change for 2.0?

This issue is something that relatively often bites people I think, so would be good to change, either as a deprecation or a 2.0 change.

**Comment by topper-123** [comment]
To add: #35967 is the issue listed in #30228, #43999 is a the same issue, but it's open, so may be the one intended for discussion about this.
